datachannel.js

```diff
@@ -0,0 +1,247 @@
+module.exports = DataChannel
+
+var inherits = require('inherits')
+var stream = require('readable-stream')
+
+var MAX_BUFFERED_AMOUNT = 64 * 1024
+
+inherits(DataChannel, stream.Duplex)
+
+function DataChannel (peer, opts) {
+  var self = this
+
+  opts = Object.assign({
+    allowHalfOpen: false
+  }, opts)
+
+  stream.Duplex.call(self, opts)
+
+  self.peer = peer
+  self._debug = self.peer._debug
+
+  self._chunk = null
+  self._cb = null
+  self._interval = null
+
+  self.channelName = null
+}
+
+DataChannel.prototype._setDataChannel = function (channel) {
+  var self = this
+
+  self._channel = channel
+  self._channel.binaryType = 'arraybuffer'
+
+  if (typeof self._channel.bufferedAmountLowThreshold === 'number') {
+    self._channel.bufferedAmountLowThreshold = MAX_BUFFERED_AMOUNT
+  }
+
+  self.channelName = self._channel.label
+
+  self._channel.onmessage = function (event) {
+    self._onChannelMessage(event)
+  }
+  self._channel.onbufferedamountlow = function () {
+    self._onChannelBufferedAmountLow()
+  }
+  self._channel.onopen = function () {
+    self._onChannelOpen()
+  }
+  self._channel.onclose = function () {
+    self._onChannelClose()
+  }
+  self._channel.onerror = function (err) {
+    self.destroy(makeError(err, 'ERR_DATA_CHANNEL'))
+  }
+
+  if (self.peer.connected) {
```

Comment:
Does `self._channel.readyState === 'open'` work instead of `self.peer.connected`?

Reply:
Implemented in https://github.com/RationalCoding/simple-peer/pull/2, addresses a third point in the first comment.